src/test/java/pl/bajerska/befindyourmeal/user/UserServiceTests.java

```diff
@@ -0,0 +1,118 @@
+package pl.bajerska.befindyourmeal.user;
+
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+import pl.bajerska.befindyourmeal.exception.ExceptionAdviceSet;
+import pl.bajerska.befindyourmeal.exception.InvalidUserPasswordException;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class UserServiceTests {
+    @Test
+    public void shouldNotFindUser() {
+
+
+        UserRepository repository = Mockito.mock(UserRepository.class);
+        UserServiceImpl service = new UserServiceImpl(repository);
+        Mockito.when(repository.findByUsername("zofia.bajerska")).thenReturn(null);
+        String name = new String("zofia.bajerska@gmail.com");
+
+        User user = service.findByUsername(name);
+
+        assertNull(user);
+    }
+
+    @Test
+    public void shouldFindUser() {
+
+        UserRepository repository = Mockito.mock(UserRepository.class);
+        UserServiceImpl service = new UserServiceImpl(repository);
```

Comment:
You can crate a service in @Before-annotated method: https://www.baeldung.com/junit-before-beforeclass-beforeeach-beforeall